ginivision/src/main/java/net/gini/android/vision/internal/qrcode/QRCodeDetectorImpl.java

```diff
@@ -0,0 +1,71 @@
+package net.gini.android.vision.internal.qrcode;
+
+import static net.gini.android.vision.internal.qrcode.QRCodeDetectorHandler.DETECT_QRCODE;
+
+import android.os.HandlerThread;
+import android.os.Message;
+import android.os.Process;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import net.gini.android.vision.internal.util.Size;
+
+import java.util.List;
+
+/**
+ * Created by Alpar Szotyori on 08.12.2017.
+ *
+ * Copyright (c) 2017 Gini GmbH.
+ */
+
+class QRCodeDetectorImpl implements QRCodeDetector {
+
+    private final QRCodeDetectorHandler mHandler;
+    private final HandlerThread mHandlerThread;
+    private Listener mListener;
+
+    QRCodeDetectorImpl(@NonNull final QRCodeDetectorTask qrCodeDetectorTask) {
+        mHandlerThread = new HandlerThread("QRCodeDetectorThread",
+                Process.THREAD_PRIORITY_BACKGROUND);
+        mHandlerThread.start();
+        mHandler = new QRCodeDetectorHandler(mHandlerThread.getLooper(), qrCodeDetectorTask);
+    }
+
+    @Override
+    public void detect(@NonNull final byte[] image, @NonNull final Size imageSize,
+            final int rotation) {
+        if (mListener == null) {
```

Comment:
Here is `mHandler` not `mListener`, right?

Reply:
If there is no Listener I avoid processing any images. (If no one is interested why do the work? 😄). The handler cannot be null.

I will add a comment to clarify this.